deisctl/config/config_test.go

```diff
@@ -2,11 +2,124 @@ package config
 
 import (
 	"encoding/base64"
+	"fmt"
 	"io/ioutil"
 	"os"
 	"testing"
 )
 
+type store struct {
+	key   string
+	value string
+}
+
+type mockClient struct {
+	expected []store
+}
+
+func (m mockClient) Get(key string) (value string, err error) {
+	for _, expect := range m.expected {
+		if expect.key == key {
+			return expect.value, nil
+		}
+	}
+	return "", fmt.Errorf("%s does not exist", m.expected)
+}
+
+func (m mockClient) Set(key, value string) (returnedValue string, err error) {
+	for _, expect := range m.expected {
+		if expect.key == key {
+			return value, nil
+		}
+	}
+	return "", fmt.Errorf("%s does not exist", m.expected)
+}
+
+func (m mockClient) Delete(key string) (err error) {
+	for _, expect := range m.expected {
+		if expect.key == key {
+			return nil
+		}
+	}
+	return fmt.Errorf("%s does not exist", m.expected)
+}
+
+type mockWriter struct {
+	output *string
```

Comment:
Instead of using a string here, would it be better to use a [`bytes.Buffer`](http://golang.org/pkg/bytes/#Buffer) instead? That way we don't have to do ugly type casting (which causes a [full copy of the buffer into the string](http://stackoverflow.com/a/9649061)) and instead just use [`io.Copy`](http://golang.org/pkg/io/#Copy).